diff --git a/etc/pacman.conf.in b/etc/pacman.conf.in
index 53071e52..e6789207 100644
--- etc/pacman.conf.in
+++ etc/pacman.conf.in
@@ -31,7 +31,7 @@ Architecture = auto
 
 # Misc options
 #UseSyslog
-#Color
+Color
 #TotalDownload
 CheckSpace
 #VerbosePkgLists
@@ -60,6 +60,9 @@
 # uncommented to enable the repo.
 #
 
+[dkp-libs]
+Server = http://downloads.devkitpro.org/packages
+
 # An example of a disabled remote package repository with multiple servers
 # available. To enable, uncomment the following lines. You can add preferred
 # servers immediately after the header and they will be used before the
