diff --git a/lib/libalpm/add.c b/lib/libalpm/add.c
index 0beed01c..860d0803 100644
--- lib/libalpm/add.c
+++ lib/libalpm/add.c
@@ -111,7 +111,7 @@ static int perform_extraction(alpm_handle_t *handle, struct archive *archive,
 {
 	int ret;
 	struct archive *archive_writer;
-	const int archive_flags = ARCHIVE_EXTRACT_OWNER |
+	const int archive_flags = /*ARCHIVE_EXTRACT_OWNER |*/\
 	                          ARCHIVE_EXTRACT_PERM |
 	                          ARCHIVE_EXTRACT_TIME |
 	                          ARCHIVE_EXTRACT_UNLINK |
@@ -656,7 +656,7 @@ static int commit_single_pkg(alpm_handle_t *handle, alpm_pkg_t *newpkg,
 int _alpm_upgrade_packages(alpm_handle_t *handle)
 {
 	size_t pkg_count, pkg_current;
-	int skip_ldconfig = 0, ret = 0;
+	int skip_ldconfig = 1, ret = 0;
 	alpm_list_t *targ;
 	alpm_trans_t *trans = handle->trans;
 
diff --git a/lib/libalpm/remove.c b/lib/libalpm/remove.c
index ffe92518..c21072f5 100644
--- lib/libalpm/remove.c
+++ lib/libalpm/remove.c
@@ -751,6 +751,7 @@ int _alpm_remove_packages(alpm_handle_t *handle, int run_ldconfig)
 	alpm_trans_t *trans = handle->trans;
 	int ret = 0;
 
+	run_ldconfig = 0;
 	pkg_count = alpm_list_count(trans->remove);
 	targ_count = 1;
 
diff --git a/lib/libalpm/util.c b/lib/libalpm/util.c
index 35fc7f41..c88f6a26 100644
--- lib/libalpm/util.c
+++ lib/libalpm/util.c
@@ -22,6 +22,10 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef __APPLE__
+#define MSG_NOSIGNAL 0
+#endif
+
 #include <stdlib.h>
 #include <unistd.h>
 #include <ctype.h>
diff --git a/scripts/pacman-key.sh.in b/scripts/pacman-key.sh.in
index 5bf6c7a1..ad84368e 100644
--- scripts/pacman-key.sh.in
+++ scripts/pacman-key.sh.in
@@ -40,7 +40,7 @@ FINGER=0
 IMPORT=0
 IMPORT_TRUSTDB=0
 INIT=0
-KEYSERVER=''
+KEYSERVER='hkp://keys.gnupg.net'
 LISTKEYS=0
 LISTSIGS=0
 LSIGNKEY=0

